[zyng233] iP

zyng233 opened the pull request:
### **PopiPro**

>  “It is not necessary to do extraordinary things to get extraordinary results.” – Warren Buffett [(source)](https://dansilvestre.com/productivity-quotes/)

PopiPro is designed to help you manage your tasks efficiently. Here's what you need to know:

- text-based
- easy to use
- FAST to use

All you need to do is,

1.  Download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html)
2.  Double-click the downloaded file
3.  Add your tasks
4.  Let it manage your tasks for you 😃

It's all **FREE**!!!

Features:

- [x] Tasks Management
- [ ] Deadline Management (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Popi {
    public  static void main(String[] args) {
        App.launch(PopiApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

zyng233 committed:
level 0

zyng233 committed:
level 0

zyng233 committed:
level 0

zyng233 committed:
level 0

zyng233 committed:
level 0

zyng233 committed:
level 4

zyng233 committed:
level 4

zyng233 committed:
level 4

zyng233 committed:
A-TextUiTesting

zyng233 committed:
Level 5

zyng233 committed:
Level 6

zyng233 committed:
Level 7

zyng233 committed:
Merge Level-7 into master

zyng233 committed:
Level 8

zyng233 committed:
extract the classes and command

zyng233 committed:
correct the delete function

zyng233 committed:
correct the delete function

zyng233 committed:
correct the delete function

zyng233 committed:
Merge branch 'branch-Level-7'

zyng233 committed:
Merge branch 'master' of https://github.com/zyng233/ip

zyng233 committed:
Merge branch 'branch-Level-8' of https://github.com/zyng233/ip into branch-Level-8

zyng233 committed:
Merge branch 'branch-Level-8'

zyng233 committed:
Merge branch-A-MoreOOP into master, resolving conflicts

zyng233 committed:
Correct the name of InvalidTimeFormatException

zyng233 committed:
Organize into packages

zyng233 committed:
Merge branch 'add-gradle-support'

zyng233 committed:
Use Gradle to build and run

zyng233 committed:
Add JUnit test

zyng233 committed:
Add JavaDoc comments

zyng233 committed:
Add CodingStandard

zyng233 committed:
Add find function

zyng233 committed:
Save local changes before merging branch-A-CodingStandard

zyng233 committed:
Merging branch-A-CodingStandard

zyng233 committed:
Merging Level-9

zyng233 committed:
Adjust the format after merging

zyng233 committed:
Checkstyle using gradle

slye20 commented on src/main/java/popi/utils/DateTimeUtils.java:
```diff
@@ -0,0 +1,37 @@
+     * @return LocalDateTime object parsed from the string.
+     * @throws InvalidTimeFormatException If the string is not in the correct format.
+     */
+    public static LocalDateTime parseDataTime(String dateTime) throws InvalidTimeFormatException {
+        try {
+            return LocalDateTime.parse(dateTime, INPUT);
+        } catch (DateTimeParseException e) {
+            throw new InvalidTimeFormatException("Please provide valid time format in yyyy-MM-dd HHmm");
+        }
+    }
```
Would a more accurate name be parseDateTime?

slye20 commented on src/main/java/popi/task/TaskList.java:
```diff
@@ -0,0 +1,118 @@
+     */
+    public Task publicMarkTask(int index) throws PopiException {
+        return markTask(index);
+    }
```
Would it better to public your markTask intead of creating a new method?

I noticed the same issue in several other places too.

slye20 commented on src/main/java/popi/task/TaskManager.java:
```diff
@@ -0,0 +1,91 @@
+import java.util.List;
+
+public class TaskManager {
+    private final String path = "./data/popi.txt";
```
Perphaps you could change it to a constant name like PATH

slye20 commented on src/main/java/popi/command/AddCommand.java:
```diff
@@ -0,0 +1,72 @@
+import popi.exception.InvalidTimeFormatException;
+import popi.exception.PopiException;
+import popi.exception.UnknownCommandException;
+import popi.task.*;
```
Perhaps it would be beter to list the imported classes explicitly.

slye20 reviewed (commented):
LGTM👍. Just a few suggestions regarding variable and method naming.

zyng233 committed:
adding JavaFX

zyng233 committed:
Refactor file handling in TaskManager

Refactor file handling in TaskManager to improve reliability.

The TaskManager class handles file operations for loading and saving
tasks. Previously, the file creation and directory handling lacked
robust error checking.

This refactor includes:
* Adding assertions for directory and file creation
* Verifying lines read from the file are non-null and non-empty
* Ensuring tasks are not null before saving

This enhances reliability and helps catch issues early in development.

zyng233 committed:
Refactor File Handling in TaskManager

The file handling logic in the TaskManager class has been refactored to improve reliability and robustness.

Adding assertions to verify successful directory and file creation.
Validating that lines read from files are non-null and non-empty.
Ensuring that tasks are not null before saving.

zyng233 committed:
Refactor task display methods to use Java Streams

Refactor the showMatchingTasks and showTasks methods to leverage Java Streams for more concise and readable code.

Replacing loops with Stream API operations in showMatchingTasks.
Using Collectors.joining() to format task list output in showTasks.
This change enhances readability and aligns with modern Java practices.

zyng233 committed:
Merge pull request #1 from zyng233/branch-A-Assertions

Implement Assertions for TaskManager File Handling

zyng233 committed:
Merge branch 'master' into branch-A-CodeQuality

zyng233 committed:
Merge pull request #2 from zyng233/branch-A-CodeQuality

Refactor Code Quality: Improve Readability and Maintainability

zyng233 committed:
Merge branch 'master' into branch-A-Streams

zyng233 committed:
Merge pull request #3 from zyng233/branch-A-Streams

Enhance Code with Java Streams API for Improved Performance

zyng233 committed:
Add CI workflow configuration with GitHub Actions

Set up Continuous Integration (CI) using GitHub Actions. The workflow
defined in .github/workflows/ci.yml builds the project using Gradle and
runs tests on each push and pull request to the master/main branch.

This setup includes:
* Checking out the code from the repository
* Setting up the JDK
* Building the project with Gradle
* Running the tests

This ensures that the code is built and tested automatically, helping
to maintain code quality and catch issues early in the development cycle.

zyng233 committed:
Add A-Personality to Popi: Establish a Unique, Sophisticated Persona

Implemented "A-Personality" for Popi, characterized by a futuristic
demeanor. This new personality aims to enhance user engagement and
create a more refined interaction experience.

Updated Popi's responses with engaging and nuanced phrases to match
this new persona. The changes are designed to improve conversational
flow and make interactions with Popi feel more dynamic and tailored.

zyng233 committed:
Merge pull request #5 from zyng233/branch-A-Personality

Add A-Personality to Popi: Establish a Unique, Sophisticated Persona

zyng233 committed:
Add User Guide for Popi

Added a comprehensive user guide for Popi, outlining essential features
and usage instructions. The guide covers task management commands,
customization options, and frequently asked questions. It provides clear
instructions for interacting with Popi, including commands for adding,
marking, and deleting tasks.

zyng233 committed:
Add User Guide for Popi

Added a comprehensive user guide for Popi, outlining essential features
and usage instructions. The guide covers task management commands,
customization options, and frequently asked questions. It provides clear
instructions for interacting with Popi, including commands for adding,
marking, and deleting tasks.

zyng233 committed:
Change Popi screenshot

Change the Ui.png to show full GUI window

zyng233 committed:
Update README.md

Correct the format of README

zyng233 committed:
Update README with Acknowledgments for JavaFX Tutorial Code

Added an acknowledgment section to the README to credit the use and
modification of code obtained from the JavaFX tutorial by SE-EDU guides.

The following components have been adapted from the tutorial:
- Classes: Main, MainWindow, DialogBox, Launcher
- FXML Files: DialogBox.fxml, MainWindow.fxml

This addition provides proper attribution and thanks to the SE-EDU guides
for their instructional resources.

zyng233 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	docs/README.md

zyng233 committed:
Update README with Acknowledgments

Added an acknowledgment section to the README to credit the use and
modification of code obtained from the JavaFX tutorial by SE-EDU guides.

The following components have been adapted from the tutorial:
- Classes: Main, MainWindow, DialogBox, Launcher
- FXML Files: DialogBox.fxml, MainWindow.fxml

This addition provides proper attribution and thanks to the SE-EDU guides
for their instructional resources.

zyng233 committed:
Fix image path for GitHub Pages

Updated the image path in README.md to ensure that Ui.png displays
correctly on GitHub Pages. Adjusted relative path to match the file
structure and verified URL accessibility.

zyng233 committed:
Edit the README.md

Correct some mistakes in the description
to make it more clean

zyng233 committed:
Solve the conflict in README.md

zyng233 committed:
Change the UnknownCommandException

zyng233 committed:
Add more description to help

zyng233 committed:
Refactor code for improved consistency

-Reformatted lines to adhere to the
character limit
-Removed redundant imports
